### PR TITLE
Creates a native accessibility experience for Radio widget

### PR DESCRIPTION
**Pull request by Myranae**
## Summary:

Goal -

The goal of these changes is to provide a native accessibility experience for users employing screen readers and
related technology. A native accessibility experience is always preferred over using ARIA attributes, so this PR makes
that change for the Radio widget.

Concerns -

1) Because we are using native radio input, we were unable to deselect an option with VoiceOver on single select questions originally. This is the native radio experience, but sighted users would be able to deselect with the visually available options. To make the experiences the same, during the review process we replaced onChange with onClick so that with every click, the checked prop could be updated. The onChange attribute was switched to an empty anonymous function. This fixed the issue regarding a difference of experience for sighted versus non-sighted users.

2) The visual button options do not reflect the value of checked via props anymore as the aria-checked prop was removed. We circumvented this for tests needing this info by testing the value of checked on the radio input elements.

(This PR is slightly long due to the nature of the change which required a lot of refactoring of related tests and snapshots.)

Issue: LP-12684

## Test plan:

- Use VoiceOver to see that each option is presented in a clean, clear way
- VoiceOver can be turned on by pressing the power button three times (on a Mac laptop)
- Here is a confluence page for Getting started with a Screen Reader: https://khanacademy.atlassian.net/wiki/spaces/AC/pages/1646624813/Getting+started+with+a+Screen+Reader
- When using screen reader, clicking a choice should update what is seen visually and the associated screen reader label with each option
- Clicking a new choice should change the visual representation of which option is selected and also update the associated screen reader labels 
- Utilizing only the visually available choices should be the same experience as it was before: each option can be deselected, one option can be selected at a time for single select questions, multiple options can be selected for multi-select questions, scoring works as usual

### CODE REVIEW

**Comment by Myranae** [comment]
Working on resolving changes and adding changeset

**Comment by Myranae** [comment]
Not sure why the code coverage report says coverage will decrease, but will look into that on Monday. Would love any suggestions on how to fix this.

File: packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx [hunk 1]
```diff
@@ -69,9 +74,9 @@ describe("base-radio", () => {
 
         // Assert
         expect(
-            screen.getByRole("checkbox", {name: "Select Choice E"}),
+            screen.getAllByRole("button", {hidden: true})[4],
```
Comment (handeyeco) [inline packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:77]: I see this throughout the PR, but `{hidden: true}` seems like a red flag - especially when we're replacing a named element like in this case. One of the goals of Testing-Library is to recreate the behavior of a user. Take this example:

1. User sees a checkbox labeled "Select Choice E"
2. There's n invisible buttons and we want button 5

The first one describes something a user would interact with and the other relies heavily on the structure of code. I'm sure you thought of this when you made the change, but I would ask that we try again to see if we can do it closer to how it was originally; if we can't, that might be a sign that things aren't as accessible as we think.

File: packages/perseus/src/widgets/__tests__/group_test.jsx [hunk 2]
```diff
@@ -21,6 +21,12 @@ describe("group widget", () => {
         jest.spyOn(Dependencies, "getDependencies").mockReturnValue(
             testDependencies,
         );
+        jest.useFakeTimers("modern");
+        jest.setSystemTime(Date.parse("04 Dec 1995 00:12:00 GMT"));
+    });
+
+    afterEach(() => {
+        jest.useRealTimers();
```
Comment (handeyeco) [inline packages/perseus/src/widgets/__tests__/group_test.jsx:29]: Are we sure we need these in all these tests? And if so, why do we need them now when we didn't need them before?

File: packages/perseus/src/widgets/radio/choice.jsx [hunk 3]
```diff
@@ -168,6 +185,26 @@ function Choice(props: ChoicePropsWithForwardRef): React.Node {
                     opacity: showDimmed ? 0.5 : 1.0,
                 }}
             >
+                <div className="perseus-sr-only">
+                    <input
+                        type={multipleSelect ? "checkbox" : "radio"}
+                        id={choiceId}
+                        name={questionId}
```
Comment (handeyeco) [inline packages/perseus/src/widgets/radio/choice.jsx:192]: Are we sure we need both `questionId` and `choiceId`?

File: .changeset/chatty-carrots-kiss.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@khanacademy/perseus": patch
```
Comment (handeyeco) [inline .changeset/chatty-carrots-kiss.md:2]: This is an intentional, user-facing issue, so it probably should be more than a patch. A patch would be something like documentation change or a bugfix (correcting an unintentional, user-facing issue).

**Review by handeyeco — COMMENTED** [review]
Thanks for looking at this. This isn't a full review, just first impressions. Tamara and I are going to walk through the code together since it's a larger PR.

File: packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx (hunk 1 above)
Comment (kevinbarabash) [reply packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:77]: Not being able to `getByRole("checkbox", ...)` worries me as well.  The wonder-blocks form elements weren't designed to allow this level of customization, but... if we want to wonder-blocks-ify perseus while maintaining the design of the radios and checkboxes, then I think we should look at what changes to wonder-blocks-form would be necessary to support that.  If there are time constraints about when we need to complete this accessibility work by then we may not have time to update wonder-blocks-form at the moment.

File: packages/perseus/src/widgets/__tests__/group_test.jsx (hunk 2 above)
Comment (kevinbarabash) [reply packages/perseus/src/widgets/__tests__/group_test.jsx:29]: In webapp, I recently changed the default to be `useRealTimers()`.  We should probably consider doing the same for perseus as well.  It makes the tests run a bit slower, but I think a fair amount of tests require it anyways and we'll need to switch whenever we upgrade `@testing-library/user-events`.

Comment (kevinbarabash) [reply packages/perseus/src/widgets/__tests__/group_test.jsx:29]: I don't think we need to change the default as part of this PR.  I can do that in a separate PR.

File: packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx (hunk 1 above)
Comment (jeanettehead) [reply packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:77]: @kevinbarabash for content I think that `getByRole("checkbox"...` would work here to reach the input experience, but she's testing the mouse interface button.

As far as incorporating the wonder blocks radio - I think that's a great plan for the future but that wasn't the plan for this piece of work - this was taking what we already had and improving the a11y interface. We need to release and update perseus in the near future and I think using the wonder blocks component might trigger conversations and work that would push that release out.

Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:77]: Will confirm clicking input (visually hidden screen reader choice) is the same as clicking the button (visually available but ARIA hidden choice), and then I will change the test back to accessing the content by checkbox.

File: packages/perseus/src/widgets/__tests__/group_test.jsx (hunk 2 above)
Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/group_test.jsx:29]: This is part of the workaround I developed for creating unique IDs. Initially I wanted to use useUniqueId, but we ran into some errors that we couldn't debug. The workaround creates IDs based on DateTime. This caused issues for snapshot testing, so using the FakeTimer allows us to bypass that issue. I will add this explanation to the code as a comment and point it out in the PR summary as well.

File: packages/perseus/src/widgets/radio/choice.jsx (hunk 3 above)
Comment (Myranae) [reply packages/perseus/src/widgets/radio/choice.jsx:192]: Will try putting the input inside the label and see if that works. If so, will remove all the choiceIds.

File: .changeset/chatty-carrots-kiss.md (hunk 4 above)
Comment (Myranae) [reply .changeset/chatty-carrots-kiss.md:2]: Will update to be minor.

**Comment by nixterrimus** [comment]
I agree with the others on a couple of nits, but on the whole this is great work. I know you were worried about the size of the change- after looking at it the core of what's changing is relatively small here and there's lots of tests to support that. I think the size is ok for this kind of change.

File: .changeset/chatty-carrots-kiss.md (hunk 4 above)
Comment (Myranae) [reply .changeset/chatty-carrots-kiss.md]: Updated to minor. Resolving :)

File: packages/perseus/src/widgets/radio/choice.jsx (hunk 3 above)
Comment (Myranae) [reply packages/perseus/src/widgets/radio/choice.jsx]: Putting the input inside the label seemed to work the same as this way, so removed choiceId and all the places that used it and moved input inside label.

File: packages/perseus/src/widgets/__tests__/group_test.jsx (hunk 2 above)
Comment (kevinbarabash) [reply packages/perseus/src/widgets/__tests__/group_test.jsx:29]: I'll try to dig into the issue with `useUniqueId*` sometime this week.

Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/group_test.jsx]: Thanks, Kevin!

Also, I fixed the tests so setSystemTime is added only to tests that need it. I looks like tests by default use Fake timers, so I just removed all the references to setting real or fake timers. If we change the default to be real timers, I can go back and add use fake timers to the tests that need it.

File: packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx (hunk 1 above)
Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx]: @handeyeco @kevinbarabash Okay, I went through and removed the hidden button references. I exchanged them with references to the radio buttons that I added. Some of my other tests show that clicking on either should produce the same results (except if you try to de-select an option by clicking the radio button; doesn't work because natively that doesn't work). I tried to have it match how it was originally as closely as possible.

Comment (handeyeco) [reply packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx]: > except if you try to de-select an option by clicking the radio button; doesn't work because natively that doesn't work

@Myranae so the behavior will be different for sighted users (cannot deselect radio) vs non-sighted users (can deselect radio)?

File: packages/perseus/src/widgets/__tests__/radio/choice_test.jsx [hunk 5]
```diff
@@ -133,30 +99,54 @@ describe("choice", () => {
...
         });
 
         // Assert
-        expect(screen.getByText("Incorrect")).toBeVisible();
+        expect(screen.getByText("Incorrect (selected)")).toBeVisible();
     });
+});
 
-    it("shows incorrect selected when in review mode", () => {
+describe("choice button", () => {
```
Comment (handeyeco) [inline packages/perseus/src/widgets/__tests__/radio/choice_test.jsx:116]: nit: it's already pretty clear that you want to test the button vs the checkbox. I just think since it is pretty atypical to have two elements controlling one control that it would be worth to have a comment explaining that this is meant to test 1/2 of those elements and the other tests test 2/2.

File: packages/perseus/src/widgets/radio/base-radio.jsx [hunk 6]
```diff
@@ -243,13 +243,25 @@ function BaseRadio(props: Props): React.Node {
     const shouldShowInstructions = !sat;
 
     const responsiveClassName = css(styles.responsiveFieldset);
+    // Note(TB): Each set of questions needs its own unique name to be
+    // considered a radio group; this is what allows only one choice at
+    // a time to be selected for single select questions. Tried
+    // useUniqueIdWithMock from WB, but encountered errors we could not fix.
```
Comment (handeyeco) [inline packages/perseus/src/widgets/radio/base-radio.jsx:249]: Since you ended up not using `useUniqueIdWithMock`, I think you can remove a lot of code from the tests such as `RenderStateRoot` or whatever and all the Jest timer business.

File: packages/perseus/src/widgets/radio/base-radio.jsx [hunk 7]
```diff
@@ -243,13 +243,25 @@ function BaseRadio(props: Props): React.Node {
...
+    // When fixed, can replace generateQuestionId with the WB function.
+    let id = 0;
+    const generateQuestionId = () => {
+        return `questionId-${Date.now()}-${id++}`;
+    };
+    const questionId = generateQuestionId();
```
Comment (handeyeco) [inline packages/perseus/src/widgets/radio/base-radio.jsx:255]: Declaring these in the body of the component won't have the effect you're expecting. Think about the render flow:

1. `BaseComponent` is called
2. `id` is defined as 0
3. `generateQuestionId` is declared
4. `generateQuestionId` is called which sets `questionId` to `"questionId-1668630158-1"`
5. `BaseComponent` is called again
6. `id` is defined as 0
7. `generateQuestionId` is declared
8. `generateQuestionId` is called which sets `questionId` to `"questionId-1668630180-1"`

You'll need to pull lines 251-254 out of the body.

---

Also I know I recommended this code, but it's important to note that this is kind of a gross hack. If I didn't recommend it, I'd probably make a case against it 😂 One reason is that it's possible for `id` to grow really big and reach a max number...in practice this would be unlikely. We could safeguard against this with the modulo operator, but I think that might be overkill since  you already marked this as a temporary hack. (NAN on this paragraph, just wanted to take blame.)

File: packages/perseus/src/widgets/radio/choice.jsx [hunk 8]
```diff
@@ -168,6 +180,25 @@ function Choice(props: ChoicePropsWithForwardRef): React.Node {
...
+                                // If we're checking a crossed-out option, let's
+                                // also uncross it.
+                                sendChange({
+                                    checked: !checked,
+                                    crossedOut: false,
+                                });
+                            }}
+                            disabled={disabled}
+                        />
+                        {a11yText} &nbsp; {content}
```
Comment (handeyeco) [inline packages/perseus/src/widgets/radio/choice.jsx:199]: Just as a learning opportunity (NAN), these are some options to avoid the `&nbsp;`

```suggestion
                        {`${a11yText} ${content}`}
```
```suggestion
                        {a11yText + " " + content}
```

File: packages/perseus/src/widgets/radio/choice-icon/sat-choice-icon.jsx [hunk 9]
```diff
@@ -87,8 +79,7 @@ function SATChoiceIcon(props: SATChoiceIconProps): React.Node {
                 style={{backgroundColor, borderColor, borderRadius}}
             />
             <div style={{color}} className={css(styles.letter)}>
-                <span className="perseus-sr-only">{a11yText}</span>
-                <span aria-hidden="true">{letter}</span>
+                <span>{letter}</span>
```
Comment (handeyeco) [inline packages/perseus/src/widgets/radio/choice-icon/sat-choice-icon.jsx:82]: This seems like a change in behavior and I wonder if it was intentional. Before the SAT choice would read "Choice A (unchecked) this is a choice" or whatever, now it looks like it will read "A Choice A (unchecked) this is a choice"...since the letter used to be hidden to screen readers.

If this is intentional, no worries. Did we check SAT with a screen reader?

**Review by handeyeco — CHANGES_REQUESTED** [review]
This is a big change, thank you for taking this on. Not much feedback really, just wondering if we could remove some of the leftover changes from trying to use WB and a possible bug by defining a function in a component.

File: packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx [hunk 10]
```diff
@@ -69,9 +74,11 @@ describe("base-radio", () => {
 
         // Assert
         expect(
-            screen.getByRole("checkbox", {name: "Select Choice E"}),
+            screen.getByRole("radio", {
+                name: "(Choice E) None of the above",
+            }),
         ).toBeInTheDocument();
-        expect(screen.getByText("None of the above")).toBeInTheDocument();
+        expect(screen.getAllByText("None of the above")).toHaveLength(2);
```
[the remark this answers is not in the document]
Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:87]: I updated the second expect statement to access the label text of the input and to confirm that is in the document. I think the length check is testing something else (making sure both options are correctly showing up as None of the above), so I left that one too but added a comment. Would it be valuable to move this to a separate test or perhaps to update the title of this test?

File: packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx [hunk 11]
```diff
@@ -201,14 +208,54 @@ describe("base-radio", () => {
...
+                screen.getByRole("radio", {name: "(Choice C) Option Gamma"}),
             );
 
             // Assert
             expect(updatedValues).toMatchObject({
                 checked: [false, false, true, false],
             });
         });
+
+        it("deselects previously selected choice", () => {
```
[the remark this answers is not in the document]
Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:226]: Yes, this is because you cannot deselect a native radio element by clicking it directly. I wanted the test to confirm that when you select a different option after having selected one already, that the previously selected choice is no longer selected.

File: packages/perseus/src/widgets/__tests__/radio/choice_test.jsx (hunk 5 above)
Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/radio/choice_test.jsx:117]: Added these comments, but let me know if they are not clear enough. Thanks for pointing this out!

File: packages/perseus/src/widgets/radio/choice.jsx (hunk 8 above)
Comment (Myranae) [reply packages/perseus/src/widgets/radio/choice.jsx:199]: Thank you! Yes, Jeanette suggested using the `&nbsp;` here; had not used it before this. I think I usually lean towards the interpolation option, though I sometimes use the concatenation option too when I don't want to type the back tick and dollar signs, lol. Is one of the options you suggested the more preferred/cleaner way for combining content? (If this were in Python, I would say more Pythonic, but I don't know if there is a similar expression for React/Javascript.)

File: packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx (hunk 1 above)
Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx]: @handeyeco yes, that part will be different. I think we talked about this with Sarah at one point during our design meetings. She said that maybe we should change that for the visual users too as you don't really expect to be able to uncheck a radio button. That's not usually how it works. In the summary for this PR, I discuss this a little. I did find a potential workaround to allow users to uncheck the native radio buttons, but it seemed a bit complex, so didn't think this was the place for that.

File: packages/perseus/src/widgets/__tests__/radio_test.jsx [hunk 12]
```diff
@@ -224,12 +231,14 @@ describe("single-choice question", () => {
...
 
             // Act
-            userEvent.tab();
-            userEvent.tab();
-            userEvent.tab();
+            userEvent.tab(); // first screen reader radio
+            userEvent.tab(); // first button
+            userEvent.tab(); // second screen reader radio
+            userEvent.tab(); // second button
+            userEvent.tab(); // third screen reader radio
```
[the remark this answers is not in the document]
Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/radio_test.jsx:235]: Visual users will have to tab twice to move between options. I don't think Jeanette has had a chance to look at the tests as I worked on these on my own. Since perseus-sr-only only visually hides options, it is still on the page. If there were some way to hide the SR content from tabbing when not in VoiceOver, that would prevent this. I feel like anything that would prevent it from being tab-able would not be able to tell if VoiceOver (or another screen reader) was on though.

File: packages/perseus/src/widgets/radio/choice-icon/sat-choice-icon.jsx (hunk 9 above)
Comment (Myranae) [reply packages/perseus/src/widgets/radio/choice-icon/sat-choice-icon.jsx:82]: Because Clickable (the outer component for the button choice) is set to aria-hidden="true", it hides everything inside, which includes ChoiceIcon and OptionStatus. This information is provided to the screen reader via the a11y text now. I also manually tested the SAT via Storybook just now and it sounds as it should. Thank you for pointing this out though!

File: packages/perseus/src/widgets/radio/base-radio.jsx (hunk 7 above)
Comment (Myranae) [reply packages/perseus/src/widgets/radio/base-radio.jsx]: Thank you so much for pointing this out! It allowed me to super simplify the IDs. It is temporary until useUniqueId works again, but it's less complicated than using the date thing.

File: packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (handeyeco) [reply packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:82]: Uhhhh, I don't have a strong opinion on splitting it out. I think it's fine as-is. Thanks for commenting it.

File: packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (handeyeco) [reply packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx:221]: I see, I think "changes selected choice" would be more accurate then since "deselected" is now kind of a loaded term in the Radio Widget.

File: packages/perseus/src/widgets/__tests__/radio_test.jsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (handeyeco) [reply packages/perseus/src/widgets/__tests__/radio_test.jsx:235]: As someone who relies heavily on keyboard navigation myself, I would consider this a blocker to merging. Could you find and propose a solution? You don't have to implement it, maybe just an idea on how we could avoid this problem.

File: packages/perseus/src/widgets/__tests__/radio_test.jsx [hunk 13]
```diff
@@ -372,8 +378,10 @@ describe("single-choice question", () => {
 
             it("should open the cross-out menu when focused and spacebar pressed", async () => {
                 // Arrange
+                jest.setTimeout(10000);
```
Comment (handeyeco) [inline packages/perseus/src/widgets/__tests__/radio_test.jsx:381]: Is this something we really need? A 10 second timeout?

File: packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx (hunk 1 above)
Comment (handeyeco) [reply packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx]: Okay, I would just double check with Sarah and Jeanette. My understanding is that having different behavior for sighted vs non-sighted users is considered a bad idea. If they say it's fine then NAN; if they say it's acceptable in the short-term, let's write a new ticket.

**Review by handeyeco — CHANGES_REQUESTED** [review]
Thanks for responding to those changes, looks like it's coming along great. I'm still hung up on the duplicate tab navigation and I think we need to find a solution for it.

File: packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/radio/base-radio_test.jsx]: Updated :)

File: packages/perseus/src/widgets/__tests__/radio_test.jsx (hunk 13 above)
Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/radio_test.jsx:381]: I received timeout errors from this test multiple times originally. I tested it again just now and it seems to be okay without, so I took it out. Will keep an eye on it though.

Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/radio_test.jsx]: Was able to take it out without a problem!

File: packages/perseus/src/widgets/radio/choice.jsx [hunk 14]
```diff
@@ -74,6 +75,11 @@ type ChoicePropsWithForwardRef = {|
     ...WithForwardRef,
 |};
 
+let id = 0;
```
Comment (jeanettehead) [inline packages/perseus/src/widgets/radio/choice.jsx:78]: A comment here would be nice linking to the fix ticket kevin is working on

Comment (Myranae) [reply packages/perseus/src/widgets/radio/choice.jsx:81]: Added! :)

File: packages/perseus/src/widgets/radio/choice.jsx [hunk 15]
```diff
@@ -168,6 +188,31 @@ function Choice(props: ChoicePropsWithForwardRef): React.Node {
                     opacity: showDimmed ? 0.5 : 1.0,
                 }}
             >
+                <div className="perseus-sr-only">
+                    <input
+                        type={multipleSelect ? "checkbox" : "radio"}
```
Comment (jeremywiebe) [inline packages/perseus/src/widgets/radio/choice.jsx:193]: Love it!!  I was wondering about this exactly as I started reviewing this PR.

File: packages/perseus/src/widgets/__tests__/radio/choice_test.jsx [hunk 16]
```diff
@@ -30,66 +30,105 @@ function renderChoice(options) {
     return render(<Choice {...overwrittenOptions} />);
 }
 
-describe("choice", () => {
-    it("renders choice content", () => {
+describe("all choice options", () => {
+    it("renders rationale", () => {
```
Comment (jeremywiebe) [inline packages/perseus/src/widgets/__tests__/radio/choice_test.jsx:34]: It took me a second to realize that some of these changes were due to re-ordering of tests.  Is there a reason you re-ordered these?  I'm not opposed to it, but usually those types of things should be done in their own PR so that the important changes in this PR aren't "muddied" by larger changes from re-organizing things.

**Review by jeremywiebe — APPROVED** [review]
Overall this looks good.  One suggestion I'd make is to try to keep PRs small and focused. There's some work to extract things into their own files and re-order tests here that will cause this PR to be bigger than necessary.

File: packages/perseus/src/widgets/__tests__/radio/choice_test.jsx (hunk 16 above)
Comment (Myranae) [reply packages/perseus/src/widgets/__tests__/radio/choice_test.jsx:34]: Thanks for pointing this out! I kept thinking I should add a comment, but I then kept forgetting. It was because I added an additional section for tests specific to the native radio options. I think we chatted about this a little actually. I wanted to create categories for the tests, so I ended up creating a category for all choices, for choice buttons, and another for choice radios. Since I needed to add the radio tests, I wanted it to be clear which tests were for which element. Unfortunately, I did not realize it would look like this in the PR.

**Review by handeyeco — APPROVED** [review]
Thanks for seeing this through the finish line
